control-plane/agents/src/bin/core/volume/clone_operations.rs

```diff
@@ -136,19 +137,24 @@ impl CreateVolumeExe for SnapshotCloneOp<'_> {
     async fn create<'a>(
         &'a self,
         context: &mut Context<'a>,
-        clone_replica: Self::Candidates,
+        clone_replicas: Self::Candidates,
     ) -> Vec<Replica> {
-        match OperationGuardArc::<ReplicaSpec>::create_ext(context.registry, &clone_replica).await {
-            Ok(replica) => vec![replica],
-            Err(error) => {
-                context.volume.error(&format!(
-                    "Failed to create replica {:?} for volume, error: {}",
-                    clone_replica,
-                    error.full_string()
-                ));
-                vec![]
+        let mut replicas = vec![];
+        for clone_replica in &clone_replicas {
```

Comment:
Where to we match up the nr of snapshot replicas and number of replicas for the clone?
Example, if we have a 3-r volume snapshot, but we want to create a 1-r volume clone, then we don't need to create 3 replica here?

Reply:
missed this check. Added now.